### PR TITLE
[Merged by Bors] - chore(RingTheory): split `AlgebraicIndependent.lean`

### PR DESCRIPTION
**Pull request by Vierkantor**
As a follow-up to splitting `RingTheory/Algebraic.lean`, also split `AlgebraicIndependent.lean`.

This PR creates the following new files:
* `AlgebraicIndependent/Defs.lean`: definition of `AlgebraicIndependent` and `IsTranscendenceBasis`
* `AlgebraicIndependent/Basic.lean`: basic results on `AlgebraicIndependent`
* `AlgebraicIndependent/TranscendenceBasis.lean`: basic results on `TranscendenceBasis`
* `AlgebraicIndependent/Adjoin.lean`: relating `AlgebraicIndependent` and `IntermediateField.adjoin`
* `AlgebraicIndependent/Transcendental.lean`: relating `AlgebraicIndependent`, `Transcendental` and `Algebraic`
* `AlgebraicIndependent/RankAndCardinality.lean`: about the rank/cardinality of `TranscendenceBasis`es

---

- [x] depends on: #19370

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by acmepjz** [comment]
Good work! I think I'll come back after #19370 is merged, as currently the diffs are too big.

File: Mathlib/RingTheory/AlgebraicIndependent/Transcendental.lean
Comment (alreadydone) [inline Mathlib/RingTheory/AlgebraicIndependent/Transcendental.lean:81]:
```suggestion
```

File: Mathlib/RingTheory/AlgebraicIndependent/Adjoin.lean
Comment (alreadydone) [inline Mathlib/RingTheory/AlgebraicIndependent/Adjoin.lean:41]: These are now unused.
```suggestion
namespace AlgebraicIndependent
```

File: Mathlib/RingTheory/AlgebraicIndependent/Adjoin.lean
Comment (alreadydone) [inline Mathlib/RingTheory/AlgebraicIndependent/Adjoin.lean:43]: Do we still need this section? It has the same scope as the `namespace` section enclosing it.

File: Mathlib/RingTheory/AlgebraicIndependent/Defs.lean
Comment (alreadydone) [inline Mathlib/RingTheory/AlgebraicIndependent/Defs.lean:50]:
```suggestion
variable {ι ι' : Type*} (R : Type*) {K A A' : Type*} (x : ι → A)
```

File: Mathlib/RingTheory/AlgebraicIndependent/Defs.lean
Comment (alreadydone) [inline Mathlib/RingTheory/AlgebraicIndependent/Defs.lean:125]: Do we need this section?

File: Mathlib/RingTheory/AlgebraicIndependent/Defs.lean
Comment (alreadydone) [inline Mathlib/RingTheory/AlgebraicIndependent/Defs.lean:152]:
```suggestion
  AlgHom.ext hx.aeval_repr
```

File: Mathlib/RingTheory/AlgebraicIndependent/Basic.lean
Comment (alreadydone) [inline Mathlib/RingTheory/AlgebraicIndependent/Basic.lean:38]: It would be good to check which of these are actually used in each file. I think `u v w` are not used in this file.

File: Mathlib/RingTheory/AlgebraicIndependent/Basic.lean
Comment (alreadydone) [inline Mathlib/RingTheory/AlgebraicIndependent/Basic.lean:45]:
```suggestion
variable {ι ι' R K A A' : Type*} {x : ι → A}
variable [CommRing R] [CommRing A] [CommRing A'] [Algebra R A] [Algebra R A']
```

File: Mathlib/RingTheory/AlgebraicIndependent/Basic.lean
Comment (alreadydone) [inline Mathlib/RingTheory/AlgebraicIndependent/Basic.lean:322]:
```suggestion
theorem AlgebraicIndependent.repr_ker (hx : AlgebraicIndependent R x) :
    RingHom.ker (hx.repr : adjoin R (range x) →+* MvPolynomial ι R) = ⊥ :=
  (RingHom.injective_iff_ker_eq_bot _).1 (AlgEquiv.injective _)
```
(untested)

File: Mathlib/RingTheory/AlgebraicIndependent/Defs.lean
Comment (alreadydone) [inline Mathlib/RingTheory/AlgebraicIndependent/Defs.lean:123]:
```suggestion
namespace AlgebraicIndependent

theorem mono {t s : Set A} (h : t ⊆ s)
    (hx : AlgebraicIndependent R ((↑) : s → A)) : AlgebraicIndependent R ((↑) : t → A) := by
  simpa [Function.comp] using hx.comp (inclusion h) (inclusion_injective h)
```

File: Mathlib/RingTheory/AlgebraicIndependent/RankAndCardinality.lean
Comment (alreadydone) [inline Mathlib/RingTheory/AlgebraicIndependent/RankAndCardinality.lean:43]: unused
```suggestion
```

File: Mathlib/RingTheory/AlgebraicIndependent/RankAndCardinality.lean
Comment (alreadydone) [inline Mathlib/RingTheory/AlgebraicIndependent/RankAndCardinality.lean:47]: A section with the same name as the file probably doesn't serve a purpose ...

File: Mathlib/RingTheory/AlgebraicIndependent/TranscendenceBasis.lean
Comment (alreadydone) [inline Mathlib/RingTheory/AlgebraicIndependent/TranscendenceBasis.lean:44]: not sure why `R` is explicit in `exists_isTranscendenceBasis`.
```suggestion
variable {ι ι' : Type*} (R : Type*) {K A A' : Type*}
variable {x : ι → A}
variable [CommRing R] [CommRing A] [CommRing A'] [Algebra R A] [Algebra R A']
```

File: Mathlib/RingTheory/AlgebraicIndependent/Transcendental.lean
Comment (alreadydone) [inline Mathlib/RingTheory/AlgebraicIndependent/Transcendental.lean:37]: I'd guess K, ι' and A' may not be used in every file ...
```suggestion
variable {ι ι' R K A A' : Type*} {x : ι → A}
variable [CommRing R] [CommRing A] [CommRing A'] [Algebra R A] [Algebra R A']
```

**Review by alreadydone — CHANGES_REQUESTED** [review]
Thanks for the effort!

File: Mathlib/RingTheory/AlgebraicIndependent/Defs.lean
Comment (Vierkantor) [reply Mathlib/RingTheory/AlgebraicIndependent/Defs.lean:123]: I'd argue this is worth keeping, both for guiding the reader and for delimiting where we have `hx` as an assumption.

**Review by alreadydone — APPROVED** [review]
Thanks 🎉 
maintainer merge

**Comment by riccardobrasca** [comment]
Thanks!

bors merge

**Pull request closed without merging**
